### PR TITLE
Python version support update

### PR DESCRIPTION
**Pull request by tdnghia98**
Story details: https://app.shortcut.com/dataiku/story/132062

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## Version 1.1.0 - Update release - 2023/04
+
+- Add support for Python versions:
```
Comment (liamlynch-data) [inline CHANGELOG.md:3]: As we did not make any code changes for these versions, perhaps it should just say "Add supported Python versions:" or "Declare support for Python versions:"?

**Comment by liamlynch-data** [comment]
The plugin version is wrong in the Makefile (still says 1.0.1) 
In fact it's probably worth getting a new version of the Makefile from the template as that would handle it automaticall

**Comment by liamlynch-data** [comment]
Testing results 
Checked 
- email macro 
- data folder file creation/delete
- dataset connection

With the following code env versions (code envs with py versions - these are on dku34 port 15200):

plugin_box-com_managed_6    2.7 - ✅ 
plugin_box-com_managed        3.6 - ✅ 
plugin_box-com_managed_1    3.7 - ✅ 
plugin_box-com_managed_2   3.8 - ✅ 
plugin_box-com_managed_3   3.9 - ✅ 
plugin_box-com_managed_4   3.10 - ❌ 
plugin_box-com_managed_5   3.11 - ❌ 

Py 3.10 and 3.11 break on email macro, folder and dataset connection - error messages:


image (1)


image (2)

```
2023-04-27 11:20:38,742 INFO Connecting to localhost (127.0.0.1) at port 40720
2023-04-27 11:20:38,742 INFO Connected to localhost (127.0.0.1) at port 40720
Traceback (most recent call last):
  File "/data/home/dataiku/dss-home/dataiku-dss-12.0.0-beta2/python/dataiku/runnables/server.py", line 26, in serve
    clazz = get_clazz_in_code(code, Runnable)
  File "/data/home/dataiku/dss-home/dataiku-dss-12.0.0-beta2/python/dataiku/base/utils.py", line 33, in get_clazz_in_code
    return get_clazz_in_file(code_file, parent_clazz)
  File "/data/home/dataiku/dss-home/dataiku-dss-12.0.0-beta2/python/dataiku/base/utils.py", line 38, in get_clazz_in_file
    mymodule = imp.load_source("dku_nomatter_module_name", code_file)
  File "/usr/local/lib/python3.10/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 719, in _load
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/tmp/tmp_folder_yyXvHynw/dku_code.py", line 2, in <module>
  File "/data/home/dataiku/dss-home/dss_design_12_test/code-envs/python/plugin_box-com_managed_4/lib/python3.10/site-packages/boxsdk/__init__.py", line 5, in <module>
    from .auth import JWTAuth, OAuth2
  File "/data/home/dataiku/dss-home/dss_design_12_test/code-envs/python/plugin_box-com_managed_4/lib/python3.10/site-packages/boxsdk/auth/__init__.py", line 5, in <module>
    from .cooperatively_managed_oauth2 import CooperativelyManagedOAuth2
  File "/data/home/dataiku/dss-home/dss_design_12_test/code-envs/python/plugin_box-com_managed_4/lib/python3.10/site-packages/boxsdk/auth/cooperatively_managed_oauth2.py", line 4, in <module>
    from .oauth2 import OAuth2
  File "/data/home/dataiku/dss-home/dss_design_12_test/code-envs/python/plugin_box-com_managed_4/lib/python3.10/site-packages/boxsdk/auth/oauth2.py", line 18, in <module>
    from ..exception import BoxOAuthException, BoxAPIException
  File "/data/home/dataiku/dss-home/dss_design_12_test/code-envs/python/plugin_box-com_managed_4/lib/python3.10/site-packages/boxsdk/exception.py", line 8, in <module>
    from .util.log import sanitize_dictionary
  File "/data/home/dataiku/dss-home/dss_design_12_test/code-envs/python/plugin_box-com_managed_4/lib/python3.10/site-packages/boxsdk/util/log.py", line 5, in <module>
    from collections import Mapping
ImportError: cannot import name 'Mapping' from 'collections' (/usr/local/lib/python3.10/collections/__init__.py)
2023-04-27 11:20:38,763 INFO Client closed
```

**Comment by liamlynch-data** [comment]
We will just remove support for 3.10 and 3.11 as these are just nice to have at the moment anyway

**Review by liamlynch-data — COMMENTED** [review]
After testing the support list needs an update - remove 3.10, 3.11
(in the config files and changelog)

The copyrights in the licence and readme need to be set or adjusted

And I made some changes to the makefile as discussed - so it does not fail is there isn't a test dir - now in line with the template as it will be https://github.com/dataiku/dss-plugin-template/pull/11

**Comment by liamlynch-data** [comment]
> After testing the support list needs an update - remove 3.10, 3.11 (in the config files and changelog)
> 
> The copyrights in the licence and readme need to be set or adjusted
> 
> And I made some changes to the makefile as discussed - so it does not fail is there isn't a test dir - now in line with the template as it will be [dataiku/dss-plugin-template#11](https://github.com/dataiku/dss-plugin-template/pull/11)

These changes made here : https://github.com/dataiku/dss-plugin-box.com/pull/6/commits/db912507c1e987426d65fb0b6b97d7b886d015f3

**Review by alexbourret — APPROVED** [review]
👍

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,62 @@
...
+plugin_id=`cat plugin.json | python -c "import sys, json; print(str(json.load(sys.stdin)['id']).replace('/',''))"`
+plugin_version=`cat plugin.json | python -c "import sys, json; print(str(json.load(sys.stdin)['version']).replace('/',''))"`
+archive_file_name="dss-plugin-${plugin_id}-${plugin_version}.zip"
+remote_url=`git config --get remote.origin.url`
+last_commit_id=`git rev-parse HEAD`
 
-plugin:
-	cat plugin.json|json_pp > /dev/null
+
+dist-clean:
```
Comment (alexbourret) [inline Makefile:9]: Could that section be moved last so that `make` defaults on `make plugin` ?

**Review by alexbourret — APPROVED** [review]
👍

**Pull request merged**